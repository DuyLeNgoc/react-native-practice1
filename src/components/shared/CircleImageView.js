/* @flow */

import React, { Component, PropTypes } from 'react';
import {
  View,
  Text,
  StyleSheet,
  Image
} from 'react-native';

export default class CircleImageView extends Component {
  render() {
    const { height, iconRatio } = this.props;
    const iconHeight = height * iconRatio;
    return (
      <View
        style={[styles.container, { height: height, borderRadius: this.props.height/2}]}>
        <Image
          style={[styles.icon, {height: iconHeight}]}
          source={this.props.imagelink}
        />
      </View>
    );
  }
}

CircleImageView.defaultProps = {
  height: 90,
  iconRatio: 0.5
};

CircleImageView.propTypes = {
  height: PropTypes.number,
  iconRatio: PropTypes.number
};

const styles = StyleSheet.create({
  container: {
    aspectRatio: 1,
    backgroundColor: 'white',
    justifyContent: 'center',
    alignItems: 'center'
  },
  icon: {
    aspectRatio: 1
  }
});
