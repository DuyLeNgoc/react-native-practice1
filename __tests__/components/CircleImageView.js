import React from 'react';
import renderer from 'react-test-renderer';
import CircleImageView from 'components/shared/CircleImageView';

it('renders correctly', () => {
  const tree = renderer.create(
		<CircleImageView
			height={120}
			imagelink={require('assets/images/check_red.png')}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
