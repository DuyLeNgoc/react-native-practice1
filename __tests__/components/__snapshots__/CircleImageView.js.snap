// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "aspectRatio": 1,
        "backgroundColor": "white",
        "justifyContent": "center",
      },
      Object {
        "borderRadius": 60,
        "height": 120,
      },
    ]
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "aspectRatio": 1,
        },
        Object {
          "height": 60,
        },
      ]
    }
  />
</View>
`;
